system-design/app-interoperability/application-package-definition.md

```diff
@@ -212,133 +393,101 @@ The expected properties for the suppported deployment types are indicated below.
 
-> **Note**  
-> Missing in the current specification are ways to define the compatibility information (resources required to run, application dependencies) as well as required infrastructure  services  such as storage, message queues/bus, reverse proxy, or authentication/authorization/accounting.
-
-## Defining configurable application variables
+## Defining configurable application parameters
 
-To allow customers to provide configuration values when installing an application, the `margo.yaml` defines the parameters and configuration sections. This gives the application vendor control over what a custom can configure when installing an application. This also describes how the workload orchestration software vendor must display these parameters to the customer to allow them to specify the values. This also describes how the workload orchestration software vendor must validate the configuration values provided by the customer before the application is installed.
+To allow customizable configuration values when installing an application, the `margo.yaml` defines the parameters and configuration sections giving the application vendor control over what can be configured when installing, or updating, an application. The [configuration](#configuration-attributes) section describes how the workload orchestration software vendor must display parameters to the user to allow them to specify the values. The [schema](#schema-attributes) section describes how the workload orchestration software vendor must validate the values provided by the user before the application is installed or updated.
 
-An example of a parameters section in the `margo.yaml` follows here:
+> **Note:** At this point the specification only deals with parameter values provided by the user as part of installing, or updating, the application. We anticipate parameter values to come from other sources, such as the device, in the future and not only from the user.
 
-```yaml
-parameters:  
-  sections:  
-    - name: General Settings  
-      settings:  
-        - Name: Greeting  
-          value: Hello  
-          targets:  
-            - key: env.APP_GREETING  
-              appliesTo: [“helm-chart”,”docker-compose”]
-          description: The greeting to use.  
-          schema: requireText  
-          dataType: string 
-        - name: Greeting Target  
-          value: World  
-          targets:  
-            - key: env.APP_TARGET  
-              appliesTo: [“helm-chart”,”docker-compose”]    
-          description: The target of the greeting.  
-          schema: requireText  
-          dataType: string 
-  schema:  
-    - name: requireText  
-      maxLength: 45  
-      allowEmpty: false  
-```
+**Parameter Attributes**
 
```

Comment:
Would it make sense to include a boolean attribute "mandatory"? This would indicate e.g., to a UI, whether a value for the parameter needs to be provided or not.

Reply:
We discussed this during the call. Optional and mandatory parameters can be defined using the schema. Each schema data type has an AllowEmpty property which can be set to true to make the parameter value optional.